### PR TITLE
[cxx-interop] Fix two issues with extending nested types across modules. 

### PR DESCRIPTION
**Pull request by zoecarver**
One fix allows extending nested records in other modules, the other fixes the same issue for namespaces.

### CODE REVIEW

**Comment by zoecarver** [comment]
@swift-ci please smoke test.

**Review by beccadax — APPROVED** [review]

**Review by egorzhdan — APPROVED** [review]
Looks good!

**Comment by zoecarver** [comment]
@swift-ci please smoke test.

**Review by hyp — APPROVED** [review]

**Comment by zoecarver** [comment]
@swift-ci please smoke test.

File: test/IDE/complete_from_clang_framework.swift [hunk 1]
```diff
@@ -122,6 +122,9 @@ func testSwiftCompletions(foo: SwiftStruct) {
 // CLANG_BAR-DAG: Decl[GlobalVar]/OtherModule[Bar]:    BAR_MACRO_1[#Int32#]{{; name=.+$}}
 // CLANG_BAR-DAG: Decl[Struct]/OtherModule[Bar]:       SomeItemSet[#SomeItemSet#]
 // CLANG_BAR-DAG: Decl[TypeAlias]/OtherModule[Bar]:    SomeEnvironment[#SomeItemSet#]
+// CLANG_BAR-DAG: Decl[TypeAlias]/OtherModule[__ObjC]: __NSConstantString[#__NSConstantString_tag#]; name=__NSConstantString
```
Comment (zoecarver) [inline test/IDE/complete_from_clang_framework.swift:125]: This patch has the side effect of exposing these types that live in __ObjC to unqualified name lookup, which I think is fine.

**Comment by zoecarver** [comment]
@swift-ci please test Windows

**Comment by zoecarver** [comment]
Looks like the windows build has been broken for a while, and the issue is unrelated, so I'm going to merge this.

**Pull request merged**